tests/integration_tests/data_steward/cdr_cleaner/cleaning_rules/remove_non_matching_participant_test.py

```diff
@@ -176,6 +176,8 @@ def setUpClass(cls):
             ehr_dataset_id=cls.ehr_dataset_id,
             validation_dataset_id=cls.validation_dataset_id)
 
+        cls.rule_instance.setup_rule(cls.client)
+
```

Comment:
You shouldn't have to call this here. It will be called as part of `default_test`. Does it fail without this?

Reply:
Yes, it fails. At line 181, `cls.rule_instance.get_sandbox_tablenames()` is called. And for that, `affected_tables` is needed.

Reply:
I think it may have to do with `self.affected_tables` being empty. Ideally in the `super().__init__()` method for the cleaning rule, `affected_tables=[table for table in CDM_TABLES if any(field['name']='person_id' for field in resources.fields_for(table))]` (pseudo code) can be a parameter.

Reply:
Oh, OK! Instead of calling the client, we can use resources.fields_for(table)). That never occurred to me. Let me try that instead.